### ISSUE TITLE
TIME_UTC and timespec_get not available on Haiku

### ISSUE THREAD

**Issue by extrowerk** [gerbera/gerbera#1202]
Compiling Gerbera on current Haiku results:

```
[  3%] Building CXX object CMakeFiles/libgerbera.dir/src/util/tools.cc.o
/Ports/Testring/gerbera/src/util/tools.cc: In function 'void getTimespecNow(timespec*)':
/Ports/Testring/gerbera/src/util/tools.cc:774:26: error: 'TIME_UTC' was not declared in this scope
     if (timespec_get(ts, TIME_UTC) == 0)
                          ^~~~~~~~
/Ports/Testring/gerbera/src/util/tools.cc:774:26: note: suggested alternative: 'UTIME_NOW'
     if (timespec_get(ts, TIME_UTC) == 0)
                          ^~~~~~~~
                          UTIME_NOW
/Ports/Testring/gerbera/src/util/tools.cc:774:9: error: 'timespec_get' was not declared in this scope
     if (timespec_get(ts, TIME_UTC) == 0)
         ^~~~~~~~~~~~
/Ports/Testring/gerbera/src/util/tools.cc:774:9: note: suggested alternative: 'timespec'
     if (timespec_get(ts, TIME_UTC) == 0)
         ^~~~~~~~~~~~
         timespec
/Ports/Testring/gerbera/src/util/tools.cc: In function 'std::filesystem::path tempName(const std::filesystem::path&, char*)':
/Ports/Testring/gerbera/src/util/tools.cc:940:23: error: 'TIME_UTC' was not declared in this scope
     timespec_get(&ts, TIME_UTC);
                       ^~~~~~~~
/Ports/Testring/gerbera/src/util/tools.cc:940:23: note: suggested alternative: 'UTIME_NOW'
     timespec_get(&ts, TIME_UTC);
                       ^~~~~~~~
                       UTIME_NOW
/Ports/Testring/gerbera/src/util/tools.cc:940:5: error: 'timespec_get' was not declared in this scope
     timespec_get(&ts, TIME_UTC);
     ^~~~~~~~~~~~
/Ports/Testring/gerbera/src/util/tools.cc:940:5: note: suggested alternative: 'timespec'
     timespec_get(&ts, TIME_UTC);
     ^~~~~~~~~~~~
     timespec
make[2]: *** [CMakeFiles/libgerbera.dir/build.make:1104: CMakeFiles/libgerbera.dir/src/util/tools.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:99: CMakeFiles/libgerbera.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
```

**Comment by whyman** [comment]
@neheb any suggestion on how to resolve this?

**Comment by neheb** [comment]
Non C11 platform...

C++17 implies C11. But that's beside the point.

I'll look into this.

**Comment by whyman** [comment]
Fixed in #1203
